### DISCUSSION TITLE
ObjectFactory Create method does not support multiple parameters.

### DISCUSSION THREAD

**Discussion by kellyethridge** [MarimerLLC/csla#1703] [Questions] [answered]
Hello,

In this issue https://github.com/MarimerLLC/cslaforum/issues/817 about creating command objects requiring a `DataPortal` create method be implemented, Rocky states that the `DataPortal` methods now support multiple parameters to improve initializing a command objects. I use an `ObjectFactory` to execute command objects, so I have to add a `Create` method. This works with zero or one parameter, but when I try multiple parameters I get an `InvalidCastException` thrown. Is multiple parameters supported for `DataPortal` methods defined in an `ObjectFactory` class? I tried with just simple data types such as just passing two integers and expecting two integers in the `Create` method just to make sure it wasn't the object I was passing. 

My calling test code is:
```
        public static async Task<int> ExecuteAsync(int employeeId, TimeOffRequestCriteria criteria)
        {
            var cmd = DataPortal.Create<GetTimeOffRequestCount>(employeeId, 1);
            cmd = await ExecuteAsync(cmd);
            return cmd.Count;
        }
```
And the `ObjectFactory` code is:
```
        [RunLocal]
        public GetTimeOffRequestCount Create(int employeeId, int extra)
        {
            var cmd = new GetTimeOffRequestCount();
            LoadProperty(cmd, GetTimeOffRequestCount.EmployeeIdProperty, employeeId);
            return cmd;
        }
```

Thank you.

**Comment by rockfordlhotka** [comment, marked as answer]
The object factory model is unchanged from earlier versions of CSLA - I've implemented the new capabilities only for the encapsulated models so far.

**Comment by hurcane** [comment]
I had a similar experience recently. I found the explanation in the books and also from older forum messages. My understanding is by no means complete, but I would summarize the situation using three use cases for making data portal calls.

- No parameters - CSLA calls the matching method that has no parameters.
- Single parameter - CSLA calls the matching method that matches the type of the parameter.
- Multiple parameters - CSLA converts the client parameters into a mobile list of "object" type and wants to call a matching method that has a single parameter.

In the multiple parameters use case, we got the invalid cast exception because we had multiple parameter signatures (i.e. overloads), including a method with a single string parameter. CSLA was attempting to convert the mobile list to the string.

We ended up using customer criteria classes to pass multiple parameters. That reduces the problem to the single parameter use case, which works just fine.
